libcaf_core/test/detail/rfc3629.cpp

```diff
@@ -162,3 +163,47 @@ TEST_CASE("invalid UTF-8 input") {
   CHECK(!valid_utf8(invalid_four_byte_9));
   CHECK(!valid_utf8(invalid_four_byte_10));
 }
+
+TEST_CASE("invalid UTF-8 input missing continuation bytes") {
+  CHECK_EQ(rfc3629::validate(invalid_two_byte_1), std::pair(0ul, true));
```

Comment:
After looking at the error on MSVC (just in case you didn't think of this already): It might be a good idea to factor out the type.

```c++
using res_t = std::pair<size_t, bool>;
```

```suggestion
  CHECK_EQ(rfc3629::validate(invalid_two_byte_1), res_t(0, true));
```

This also helps making the code a bit cleaner since we can drop the `ul` suffixes. 🙂

Reply:
Thanks a lot. I'll also shift to the new test framework to get rid of CHECK macros.